test/Autofac.Test/Features/Scanning/OpenGenericScanningRegistrationTests.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Copyright (c) Autofac Project. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Autofac.Core;
+using Autofac.Core.Lifetime;
+using Autofac.Core.Registration;
+using Autofac.Features.Scanning;
+using Autofac.Test.Scenarios.ScannedAssembly;
+using Xunit;
+
+namespace Autofac.Test.Features.Scanning
+{
+    public class OpenGenericScanningRegistrationTests
+    {
+        private static readonly Assembly ScenarioAssembly = typeof(AComponent).GetTypeInfo().Assembly;
+
+        [Fact]
+        public void WhenAssemblyIsScannedOpenGenericTypesCanBeResolved()
+        {
+            var cb = new ContainerBuilder();
+            cb.RegisterAssemblyOpenGenericTypes(typeof(AComponent).GetTypeInfo().Assembly);
+            var c = cb.Build();
+
+            Assert.NotNull(c.Resolve<RedoOpenGenericCommand<int>>());
+
+            c.AssertRegistered<RedoOpenGenericCommand<int>>();
+            c.AssertSharing<RedoOpenGenericCommand<int>>(InstanceSharing.None);
+            c.AssertLifetime<RedoOpenGenericCommand<int>, CurrentScopeLifetime>();
+            c.AssertOwnership<RedoOpenGenericCommand<int>>(InstanceOwnership.OwnedByLifetimeScope);
+        }
+
+        [Fact]
+        public void WhenOpenGenericTypesRegisteredAsSelfConcreteTypeIsService()
+        {
+            var cb = new ContainerBuilder();
+            cb.RegisterAssemblyOpenGenericTypes(typeof(AComponent).GetTypeInfo().Assembly)
+                .AsSelf();
+            var c = cb.Build();
+
+            Assert.NotNull(c.Resolve<RedoOpenGenericCommand<int>>());
+
+            c.AssertRegistered<RedoOpenGenericCommand<int>>();
+        }
+
+        [Fact]
+        public void WhenFiltersAppliedNonMatchingOpenGenericTypesExcluded()
+        {
+            var cb = new ContainerBuilder();
+            cb.RegisterAssemblyOpenGenericTypes(typeof(AComponent).GetTypeInfo().Assembly)
+                .Where(t => t.Name.StartsWith("Re"))
```

Comment:
I'm not sure why there are two `Where` clauses here that overlap. Probably just need one.

Reply:
The reason for there are two `Where` clauses is that I follow the same test case in this file: https://github.com/autofac/Autofac/blob/develop/test/Autofac.Test/Features/Scanning/ScanningRegistrationTests.cs#L76

Reply:
That's just... weird. I don't know why that'd make any difference. I think in this one we can remove the first `Where` clause.

Reply:
Thanks @tillig , just updated my PR to resolve it.